### PR TITLE
NEW: Check Thirdparty name

### PR DESCRIPTION
**Pull request by milenmk**
# New [*Check Thirdparty name*]

When adding a new client, check if a Third Party name already exists and throw an error if so.

### CODE REVIEW

**Comment by atm-maxime** [comment]
🤔 two thirdparties can have the same name...

File: htdocs/societe/card.php [hunk 1]
```diff
@@ -406,6 +406,20 @@
 			setEventMessages($langs->trans("ErrorFieldRequired", $langs->transnoentities("No_Email")), null, 'errors');
 		}
 
+		if ($action == 'add') {
+			$thirdparty = GETPOST('name');
+			$sql = 'SELECT nom FROM '.MAIN_DB_PREFIX.'societe WHERE nom="'.$db->escape($thirdparty).'"';
```
Comment (eldy) [inline htdocs/societe/card.php:411]: It is frequent that 2 companies has the same name. 
Currenlty this is allowed into Dolibar but we may add an option to forget this.
You can add a test here 
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))

Comment (daraelmin) [reply htdocs/societe/card.php:411]: What about individual ?  How would you manage namesake of humain being ?
How would you manage multiple business units, parent and child society with sans name ?

Comment (milenmk) [reply htdocs/societe/card.php:411]: It is possible to add a filter to this, like 

`
if (! empty($conf->global->THIRDPARTY_NAME_MUST_BE_UNIQUE))
`

**Comment by BB2A-Anthony** [comment]
It's interesing add option

And if existe proposed to create or not

It's interesing for product

**Comment by eldy** [comment]
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolete, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.

**Pull request closed without merging**